src/Composer/Installers/HnhInstaller.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace Composer\Installers;
+
+class HnhInstaller extends BaseInstaller
+{
+    protected $locations = [
+        'module' => 'modules/{$name}/',
+        'plugin' => 'plugins/{$name}/',
+        'theme'  => 'themes/{$name}/',
+    ];
+
+    /**
+     * Format package name.
+     *
+     * For package type hnh-module, cut off a trailing '-module' if present.
+     *
+     * For package type hnh-plugin, cut off a trailing '-plugin' if present.
+     *
+     * For package type hnh-theme, cut off a trailing '-theme' if present.
+     *
+     */
+    public function inflectPackageVars($vars)
```

Comment:
@shama @jadb what tell into the account of magic names? Is this right or to use extra settings (`installer-name`)?